### DISCUSSION TITLE
Redesign

### DISCUSSION THREAD

**Discussion by Waakul** [ScratchAddons/ScratchAddons#7215] [Ideas]
### Why this would be helpful

More modern interface

### My suggestion

Redesigns by me
![Frame 28 (2)](https://github.com/ScratchAddons/ScratchAddons/assets/135821321/da9d346e-1815-458c-ace5-7d2562f5ed10)
i will develop this but we need need to discuss about this first so ill create an issue

**Comment by Hexagonl** [comment]
I would like to note I only assisted in the concept art on Discord because I was bored and it was mildly interesting, I do not feel that SA requires a redesign as of this moment. Most of the work is by Waakul, all I did was fix the color contrast and wasted space issues

**Comment by Waakul** [comment]
tbh i'm not making it look exactly like this, I'm making some changes while I'm developing it

**Comment by Waakul** [comment]
> I would like to note I only assisted in the concept art on Discord because I was bored and it was mildly interesting, I do not feel that SA requires a redesign as of this moment. Most of the work is by Waakul, all I did was fix the color contrast and wasted space issues

we really need something cool for v2.0.0 and the current design sucks so...

**Comment by Waakul** [comment]
![image](https://github.com/ScratchAddons/ScratchAddons/assets/135821321/d53c9b06-01ff-4bac-9903-6dbce06b17c5)
current progress screenshot

**Comment by Samq64** [comment]
There are links to some other settings page redesigns here: https://github.com/ScratchAddons/ScratchAddons/discussions/5103#discussioncomment-3631004

Also unfortunately we're still using a really old version of Vue and are trying to go easy on changes until it's ported to a newer version (#6272) so a proper redesign won't happen until after that but smaller changes should be fine.

**Comment by Waakul** [comment]
> There are links to some other settings page redesigns here: [#5103 (comment)](https://github.com/ScratchAddons/ScratchAddons/discussions/5103#discussioncomment-3631004)
> 
> Also unfortunately we're still using a really old version of Vue and are trying to go easy on changes until it's ported to a newer version (#6272) so a proper redesign won't happen until after that but smaller changes should be fine.

no much features, only CSS changes

**Comment by Waakul** [comment]
> There are links to some other settings page redesigns here: [#5103 (comment)](https://github.com/ScratchAddons/ScratchAddons/discussions/5103#discussioncomment-3631004)
> 
> Also unfortunately we're still using a really old version of Vue and are trying to go easy on changes until it's ported to a newer version (#6272) so a proper redesign won't happen until after that but smaller changes should be fine.

porting to vue3 wont effect css files...

**Comment by Waakul** [comment]
wait we could do vue3, mv3, and redesign all together and bump into v2.0.0

**Reply by DNin01** [reply]
Migrating to MV3 on time is more important than redesigning the settings pages, but the upgrade to Vue 3 would be a good time to simultaneously redesign the settings page.

**Reply by Waakul** [reply]
yes indeed, but i'm doing the redesign and others are doing the vue3 and mv3. so we'll probably get everything finished by june and we could bump into v2.0.0,

**Reply by BroJac5246** [reply]
> Migrating to MV3 on time is more important than redesigning the settings pages, but the upgrade to Vue 3 would be a good time to simultaneously redesign the settings page.

Vue 3 is also more important than a settings page redesign. The current settings page design works fine and we haven't even determined new features for v2, much less what the settings page needs to look like.

**Comment by Samq64** [comment]
> porting to vue3 wont effect css files...

Everything is moved around and Git doesn't track everything correctly so there might be a couple merge conflicts but I don't mind fixing those.

> wait we could do vue3, mv3, and redesign all together and bump into v2.0.0

I think that's the plan assuming it's all ready by June.

**Reply by BroJac5246** [reply]
Sure, but what's the probability that v2 will be ready or even designed by June?

**Comment by Waakul** [comment]
> > porting to vue3 wont effect css files...
> 
> Everything is moved around and Git doesn't track everything correctly so there might be a couple merge conflicts but I don't mind fixing those.
> 
> > wait we could do vue3, mv3, and redesign all together and bump into v2.0.0
> 
> I think that's the plan assuming it's all ready by June.

ik, I'll try to fix those myself later ig

**Comment by GrahamSH-LLK** [comment]
design ideas should go in a discussion

**Comment by Waakul** [comment]
@GrahamSH-LLK huh?

**Comment by Waakul** [comment]
@MaterArc told me to open an issue instead

**Reply by cobaltt7** [reply]
Mater is not a SA maintainer.

**Comment by DNin01** [comment]
I like your approach with the design. While I'm not sure if we're going to call it 2.0, the design style in your images looks cleaner and more modern. I'll have to try your pull request later.

**Comment by Waakul** [comment]


**Reply by shock59** [reply]
Putting `(no offense)` at the end of your post doesn't suddenly make it not rude if it was before - you could phrase this a lot better if you're looking for actual discussion because here it looks like you're just being mean

**Comment by cally-jt** [comment]
> @callumjt states that materarc's this design is better than mine 🤣 (no offense):
> 
> ![image](https://github.com/ScratchAddons/ScratchAddons/assets/135821321/bbbc8e50-1a76-42cc-bedf-7c879f842536)
> 
> design lang not consistent, padding not consistent, even the UI is pretty vague

Mocking someone and laughing at someone else's design is the best way for someone to hate you

**Reply by Waakul** [reply]
i just wanted other thoughts on it

**Reply by cally-jt** [reply]
The best way to get thoughts is by asking them, not laughing at someone's design

**Reply by Waakul** [reply]
I ALREADY SAID I'M NOT LAUGHING AT HIS DESIGN

**Comment by cally-jt** [comment]
Also, you said the padding was off, I literally sent a pic of mater saying he would change the padding so I don't see what your point is there

**Reply by Waakul** [reply]
ik but design lang is highly inconsistent

**Reply by cally-jt** [reply]
And? Why does that make you laugh at someone's design?

**Reply by Waakul** [reply]
i didnt laugh at his design, i laughed at the thing that you think it's better than mine

**Reply by cally-jt** [reply]
Your clearly not willing to realise people have opinions and the way you structured the sentence made it seem like you laughed at his design. Just don't use emojis like that in the future 🤦

**Reply by Waakul** [reply]
bruh

**Comment by Waakul** [comment]
@WorldLanguages so will it be implemented in v2 or should i stop working on this?

**Comment by Waakul** [comment]
Can someone give me ideas? i dont have any... or at least some minor layout changes??

**Reply by Samq64** [reply]
https://github.com/ScratchAddons/ScratchAddons/issues?q=is%3Aissue+is%3Aopen+label%3A%22scope%3A+webpages%22+label%3A%22scope%3A+design%22

**Reply by Samq64** [reply]
Out of those here are a few that don't have any linked PRs and shouldn't require too many non-CSS changes.

- #3642
- #6035
- #6363
- #6997

**Comment by Waakul** [comment]
![image](https://github.com/ScratchAddons/ScratchAddons/assets/135821321/53bc2017-bcbb-4be1-b012-6cc409795b35)
progress

**Comment by Secret-chest** [comment]
Useability and layout > style.

**Reply by DNin01** [reply]
How does this redesign concept impact the usability and layout?

**Reply by Samq64** [reply]
Note that #7214 has changed quite a bit and the screenshots are outdated but there still aren't any big layout changes. If you mean the popup design from Discord I agree it's not obvious what the tabs are (I also don't know why that was posted here instead of a separate discussion).

**Comment by Secret-chest** [comment]
> new features for v2

Scratch Addons works fine, why are we so obsessed with v2 being a major user-facing update? Of course mv3 and Vue 3 would warrant a major version bump, but it's not going to be that much.

**Comment by Waakul** [comment]
![Redesign by Waakul · Pull Request #7214 · ScratchAddons_ScratchAddons and 4 more pages - Personal - Microsoft​ Edge 3_9_2024 2_32_14 PM](https://github.com/ScratchAddons/ScratchAddons/assets/135821321/a75ec2df-8d79-4f0f-992f-b401f20a2a08)
i haven't posted any updates in this discussion so here ya go
